controllers/releaseplan/adapter.go

```diff
@@ -74,3 +76,26 @@ func (a *adapter) EnsureOwnerReferenceIsSet() (controller.OperationResult, error
 
 	return controller.ContinueProcessing()
 }
+
+// EnsureMatchingInformationIsSet is an operation that will ensure that the ReleasePlan has updated matching
+// information in its status.
+func (a *adapter) EnsureMatchingInformationIsSet() (controller.OperationResult, error) {
+	// If an error occurs getting the ReleasePlanAdmission, mark the ReleasePlan as unmatched
+	releasePlanAdmission, _ := a.loader.GetReleasePlanAdmissionFromReleasePlan(a.ctx, a.client, a.releasePlan)
```

Comment:
I don't like this. The error could be network related. You should check that is a IsNotFound error.